corehq/apps/reports/views.py

```diff
@@ -2008,6 +2010,16 @@ def mk_date_range(start=None, end=None, ago=timedelta(days=7), iso=False):
         return start, end
 
 
+def _is_location_safe_report_class(request, domain, export_hash, format):
+    cache = get_redis_client()
+
+    content = cache.get(export_hash)
+    if content is not None:
+        report_class, report_file = content
+        return report_class_is_location_safe(report_class)
+
+
+@conditionally_location_safe(_is_location_safe_report_class)
```

Comment:
Probably does not make a difference, just the decorator order would look better if this comes after all conditions are satisfied
```
@login_and_domain_required
@require_GET
@conditionally_location_safe(_is_location_safe_report_class)
```

Unless there is a reason for putting it at the top instead.